src/aap_eda/api/serializers/mixins.py

```diff
@@ -0,0 +1,27 @@
+#  Copyright 2024 Red Hat, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from django.conf import settings
+
+from aap_eda.api import exceptions as api_exc
+
+
+class SharedResourceSerializerMixin:
+    def validate_shared_resource(self):
+        if settings.DIRECT_SHARED_RESOURCE_MANAGEMENT_ENABLED:
+            view = self.context.get("view")
+            action = view.action.capitalize() if view else "Action"
+            raise api_exc.Forbidden(
```

Comment:
Is 403 the right response? I remember some previous discussion about it and I think the conclusion was that 403 is only for access rights, and things like this should be 409
```This response is sent when a request conflicts with the current state of the server.```

Reply:
The parent issue https://issues.redhat.com/browse/AAP-24178 requests the return code to be 403.